core/domain/exp_domain.py

```diff
@@ -2548,13 +3600,14 @@ def update_states_from_model(
                 versioned_exploration_states['states'], init_state_name)
         else:
             versioned_exploration_states['states'] = conversion_fn(
-                versioned_exploration_states['states'])
+                versioned_exploration_states['states'], language_code
+            )
```

Comment:
Have you tested this yet? This won't work and won't be a good thing to do for the future. If the state schema version if 50, then this will try to run `_convert_states_v50_dict_to_v51_dict(states, language_code)`, but that function doesn't take in a `language_code` argument. 

Also, adding this `language_code` argument here makes it so that every future schema migration would need to pass in the `language_code` even if it is not used. You should make a one-off `elif` statement instead like it does for `v43`.

Reply:
Yeah you are right, updated!

Reply:
I tested it and it was working fine because the maximum version will be 51. I have updated it thanks for pointing it out